### ISSUE TITLE
YOLOv5 custom trained model is confusing classes

### ISSUE THREAD

**Issue by JustBrandonLim** [ultralytics/yolov5#10031]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I am new to machine learning, and am recently trying out YOLOv5 to detect objects in a game to solve for arrow puzzles.

I have 4 classes, left, right, up and down.

up and down are detected quite accurately with a confidence of usually 0.9 ish?

However, left arrows are always detected as right arrows with an even higher confidence than actual right arrows.

Could it be due to class imbalance or the lack of samples?

![confusion_matrix](https://user-images.githubusercontent.com/93673724/199940410-c07fbbfb-a9d4-45d2-9cb3-ad1591290632.png)

![labels](https://user-images.githubusercontent.com/93673724/199940449-e5baca3e-b7ea-4c83-8c1e-a4c7bf495152.jpg)

![results](https://user-images.githubusercontent.com/93673724/199940526-19d694d3-6564-4b0f-9699-6f4c25698e54.png)

![P_curve](https://user-images.githubusercontent.com/93673724/199940740-46bff048-16b4-43b7-b8d7-7be070953b36.png)

I am training with `yolov5s.pt` and using rect as my images are 640 x 160.

```python
    train.run(
        device='0',
        imgsz=640, 
        data='data.yaml', 
        #cfg='yolo5s.yaml',
        weights='yolov5s.pt',  
        epochs=100, 
        batch=100, 
        workers=4,
        rect=True)
```

I understand that the recommended is about >= 1500 images per class and >= 10000 instances per class. But I have seen tutorials where 200 images could work as well.

**Comment by adrianholovaty** [comment]
When you do custom training with YOLOv5, the system automatically applies image augmentation — that is, altering your training images slightly, to artificially expand your number of training samples. This includes image "flipping" by default, doing a mirror image both horizontally and vertically.

In your case, it sounds like you definitely _shouldn't_ be applying the flipping augmentations to your training data, because in your particular application there's an important distinction between left and right arrows (and between up and down arrows).

You can disable these augmentations by making a custom hyperparameters file and changing the values for `flipud` and `fliplr`:

```
flipud: 0.0  # image flip up-down (probability)
fliplr: 0.0  # image flip left-right (probability)
```

Then you'll have to retrain, with this file, in order to produce a new model.

**Comment by JustBrandonLim** [comment]
> When you do custom training with YOLOv5, the system automatically applies image augmentation — that is, altering your training images slightly, to artificially expand your number of training samples. This includes image "flipping" by default, doing a mirror image both horizontally and vertically.
> 
> In your case, it sounds like you definitely _shouldn't_ be applying the flipping augmentations to your training data, because in your particular application there's an important distinction between left and right arrows (and between up and down arrows).
> 
> You can disable these augmentations by making a custom hyperparameters file and changing the values for `flipud` and `fliplr`:
> 
> ```
> flipud: 0.0  # image flip up-down (probability)
> fliplr: 0.0  # image flip left-right (probability)
> ```
> 
> Then you'll have to retrain, with this file, in order to produce a new model.

Thank you, good sir. It works perfect now with yolov5x.pt too!

Just a few questions, if let's say I were to have an image for 1366 x 768, and I trained at `imgsz=320`, how should I export my ONNX?

I also have another one (for the arrows), image is 640 x 160, trained at `imgsz=640`, how should I export my ONNX as well?

**Comment by glenn-jocher** [comment]
@JustBrandonLim glad to hear that it's working well for you now! For exporting an ONNX model with YOLOv5, the process is quite straightforward. 

For the image with the resolution 1366 x 768, trained at `imgsz=320`, you would export your ONNX by running the command:
```
python export.py --weights 'path to your trained weights' --img 320 --batch 1 --device 'cpu' --include torchscript
```

For the image with the resolution 640 x 160, trained at `imgsz=640`, the export command would be:
```
python export.py --weights 'path to your trained weights' --img 640 --batch 1 --device 'cpu' --include torchscript
```

These commands will result in the creation of two ONNX models tailored to their specific image sizes.

Don't hesitate to consult the [Ultralytics Docs](https://docs.ultralytics.com/yolov5/) for additional guidance on ONNX exporting and for any other questions that may arise.
